docs/docsite/rst/dev_guide/developing_modules_documenting.rst

```diff
@@ -273,15 +273,18 @@ Linking within module documentation
 
 You can link from your module documentation to other module docs, other resources on docs.ansible.com, and resources elsewhere on the internet. The correct formats for these links are:
 
-* ``L()`` for Links with a heading. For example: ``See L(IOS Platform Options guide,../network/user_guide/platform_ios.html).``
+* ``L()`` for links with a heading. For example: ``See L(Ansible Tower,https://www.ansible.com/products/tower).`` As of Ansible 2.10, do not use ``L()`` for relative links to other Ansible documentation. Use ``R()`` instead.
 * ``U()`` for URLs. For example: ``See U(https://www.ansible.com/products/tower) for an overview.``
+* ``R()`` for cross-references with a heading (added in Ansible 2.10). For example: ``See R(Cisco IOS Platform Guide,ios_platform_options).`` Use the RST anchor for the cross-reference.
 * ``I()`` for option names. For example: ``Required if I(state=present).``
 * ``C()`` for files and option values. For example: ``If not set the environment variable C(ACME_PASSWORD) will be used.``
 * ``M()`` for module names. For example: ``See also M(win_copy) or M(win_template).``
 
 .. note::
 
-  For modules in a collection, you can only use ``L()`` and ``M()`` for content within that collection. Use ``U()`` to refer to content in a different collection.
+  For modules in a collection, you can only use ``L()`` and ``M()`` for content within that collection. Use ``U()`` to refer to content in a different collection, or ``R()`` for cross-references to other Ansible documentation, based on the anchor for the page or section of the ``.rst`` file.
```

Comment:
A few notes about this section

* `L()` can be used for content outside the collection.  So maybe that part is better phrased as:
   > you should only use relative links with ``L()`` and ``U()`` if the content lives within the collection.
* relative links can work outside of the collection, but only if you know the directory structure outside of the collection.  In particular, that means that a relative link that works with docs-pipeline probably won't work on the collection maintainer's website and vice versa.
* `M()` and `R()` should work inside of a collection always.  Pointing outside of a collection, they can work in the docs pipeline.  They may be able to work on a collection maintainer's website if they have intersphinx setup appropriately and there's no conflicts in the ref names.

Reply:
I'm confused now. 

1. I thought U() was only for full urls, not relative links?  At least that is the example we use.

2. For L() as a relative link, is that to the collection docs folder? to..??? not sure what a relative link within a collection looks like.

3. For relative links outside a collection, that just seems problematic and I suggest we don't advertise it.  

4. M() and R()  - @acozine - I'm of two minds here - it's a good point that people building their own docsite would need to know they need intersphinx to get them working, but I'm not sure we are/should be in the business of documenting how someone builds their own docsite from module content.  I could just add a phrase after  that last sentence to say something like "These cross-references to other Ansible documentation work based on the intersphinx extention'. ?

Reply:
1. The only difference between 'U()' and 'L()' that I can see in the code is whether to include the bare link or to include a heading.  for example:
    ``` html
    <!--  U("../../collections/index.html") turns into -->
    <a href="../../collections/index.html">../../collections/index.html</a>

    <!-- L(List of collections, "../../collections/index.html") turns into -->
    <a href="../../collections/index.html">List of collections</a>
    ```
   (absolute URLS will work for both of those as well)
2. Relative links are relative to the html page.  So, if I am working on documentation in the yum module from ansible-base, I could refer to the apt module's documentation via: ```L(The apt module,apt_module.html)``` because the apt module is also in ansible-base and will therefore be placed in the same directory.  If I am working on the  yum module docs and i wanted to reference something in another collection, I could use ```L(Ansible documentation,../../../index.html)``` to reference the toplevel index page of the docs.ansible.com website.  (But on a collection owner's website, that might not be the correct location for what they want to reference).
3. Yep.  Agreed
4. I know you're really asking acozine but I just wanted to say I agree, something that says this is possible but points them elsewhere to get the details seems like a good compromise.  Although, we may want people to know that if they're only developing for publication on docs.ansible.com, they don't have to know about intersphinx?

Reply:
Okay so U() is for raw links so to speak (no title) and L() includes a title for the link.
Relative linking needs more explanation, and I think we should limit it to -use docs.ansible.com relative links only if you are in ansible base.  Use relative links in your collection only to your own collection.  Use L() or U() to link from your collection to docs.ansible.com pages outside your collection.

Reply:
Discussed in #ansible-doc and we will mention intersphinx w/ a link to the intersphinx docs for how to get M() and R() working on a separate docsite.